app/src/androidTest/java/net/aiscope/gdd_app/SampleDtoKtTest.kt

```diff
@@ -0,0 +1,28 @@
+package net.aiscope.gdd_app
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import net.aiscope.gdd_app.model.Sample
+import net.aiscope.gdd_app.network.toDto
+import org.junit.Assert
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class SampleDtoKtTest {
+
+    @Test
+    fun shouldFillExpectedFields() {
+        val sample = Sample("_id",
+            "facility",
+            "microscopist")
+
+        //This is the DTO for sending to firebase rather than storing locally
+        val dto = sample.toDto()
+
+        Assert.assertEquals(dto.id, "_id")
+        Assert.assertEquals(dto.healthFacility,"facility")
+        Assert.assertEquals(dto.microscopist, "microscopist")
+        Assert.assertTrue(dto.appVersion.matches(Regex("""^\d+\.\d+\.\d+ \([a-zA-Z0-9]+\)""")))

```

Comment:
Extract Regex object to a companion object constant?

Reply:
Extracted to top-level val instead